prog/vm/nexus.rb

```diff
@@ -64,29 +64,27 @@ def self.assemble(public_key, project_id, name: nil, size: "standard-2",
       # then we create a nic on that subnet.
       # - If the user provided neither nic_id nor private_subnet_id, that's OK, we create both.
       nic = nil
-      subnet = nil
-      if nic_id
+      subnet = if nic_id
         nic = Nic[nic_id]
         raise("Given nic doesn't exist with the id #{nic_id}") unless nic
         raise("Given nic is assigned to a VM already") if nic.vm_id
         raise("Given nic is created in a different location") if nic.private_subnet.location != location
         raise("Given nic is not available in the given project") unless project.private_subnets.any? { |ps| ps.id == nic.private_subnet_id }
 
-        subnet = nic.private_subnet
+        nic.private_subnet
       end
 
       unless nic
-        subnet = nil
-        if private_subnet_id
+        subnet = if private_subnet_id
           subnet = PrivateSubnet[private_subnet_id]
           raise "Given subnet doesn't exist with the id #{private_subnet_id}" unless subnet
           raise "Given subnet is not available in the given project" unless project.private_subnets.any? { |ps| ps.id == subnet.id }
+          subnet
         else
-          subnet_s = Prog::Vnet::SubnetNexus.assemble(project_id, name: "#{name}-subnet", location: location, allow_only_ssh: allow_only_ssh)
-          subnet = PrivateSubnet[subnet_s.id]
+          subnet = Prog::Vnet::SubnetNexus.assemble(project_id, name: "#{name}-subnet", location: location, allow_only_ssh: allow_only_ssh).subject
```

Comment:
This is a straight local respelling of some variables to clean up some scopes, right? Does it have something to do with the stated purpose of the patch?

Reply:
You're right, I moved them to a separate commit